lib/ci/bin/rails_upgrade_analyzer

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env ruby
+
+require "stringio"
+require_relative "../lib/rails_upgrade_analyzer"
+require_relative "../../deprecation_summary"
+
+module RailsUpgradeAnalyzer
+  class Listener
+
+    REPO_NAME = "Liaison-Intl/WebAdMIT"
+
+    RAILS_4_ACCEPTED = "Rails 4 Accepted"
+    RAILS_4_REJECTED = "Rails 4 Rejected"
+    RAILS_4_REBASE = "Rails 4 CI Needed (rebase master)"
+    RAILS_4_WARNING = "Rails 4 Check Deprecation Warnings"
+
+    def listen
+      puts "Initializing build listener"
+
+      login
+
+      Travis::Pro.listen(webadmit) do |stream|
+        stream.on("build:finished") do |event|
+          run(event.build)
+        end
+      end
+    end
+
+    def run_one(build_number)
+      login
+
+      build = webadmit.build(build_number)
+      run(build)
+    end
+
+    private
+
+    attr_reader :build_number
+
+    def login
+      Travis::Pro.github_auth(GithubProxy.access_token)
+    end
+
+    def run(build)
+      unless build.pull_request?
+        puts "Build #{build.number} is not a pull request. Skipping."
+        return
+      end
+
+      webadmit.session.clear_cache
+
+      puts "Analyzing PR: #{build.pull_request_number}"
+      results = analyze_build(build, "Current")
+      puts "Getting master results"
+      base = base_branch(build)
+      master_results = analyze_build(last_complete_build(base), base)
+      report_results(build, results, master_results)
+    end
+
+    def base_branch(build)
+      match = build.branch_info.match(/\A[^\s]*/)
```

Comment:
I **think** this method could be...

```ruby
def base_branch(build)
  build.branch_info.match(/\A[^\s]*/).to_s
end
```

...but this is something I just discovered recently, so I might be wrong about that.  :smile: